Client/Frontend/Widgets/BlurTableViewController.swift

```diff
@@ -0,0 +1,182 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Storage
+
+class BlurTableViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    var profile: Profile!
+    var site: Site!
+    var actions: [BlurTableViewAction] = []
+    var tableView = UITableView()
+    lazy var tapRecognizer: UITapGestureRecognizer = {
+        let tapRecognizer = UITapGestureRecognizer()
+        tapRecognizer.addTarget(self, action: #selector(BlurTableViewController.dismiss(_:)))
+        tapRecognizer.numberOfTapsRequired = 1
+        tapRecognizer.cancelsTouchesInView = false
+        return tapRecognizer
+    }()
+
+    lazy var visualEffectView : UIVisualEffectView = {
+        let visualEffectView = UIVisualEffectView(effect: UIBlurEffect(style: .Light))
+        visualEffectView.frame = self.view.bounds
+        visualEffectView.alpha = 0.90
+        return visualEffectView
+    }()
+
+    lazy var bookmarkAction : BlurTableViewAction = {
+        return BlurTableViewAction(title: NSLocalizedString("Bookmark", comment: "Context Menu Action for Activity Stream"), iconString: "action_bookmark", handler: { action in
+            let shareItem = ShareItem(url: self.site.url, title: self.site.title, favicon: self.site.icon)
+            self.profile.bookmarks.shareItem(shareItem)
+            if #available(iOS 9, *) {
+                var userData = [QuickActions.TabURLKey: shareItem.url]
+                if let title = shareItem.title {
+                    userData[QuickActions.TabTitleKey] = title
+                }
+                QuickActions.sharedInstance.addDynamicApplicationShortcutItemOfType(.OpenLastBookmark,
+                    withUserData: userData,
+                    toApplication: UIApplication.sharedApplication())
+            }
+        })
+    }()
+
+    lazy var deleteFromHistoryAction : BlurTableViewAction = {
```

Comment:
We can probably make all these properties `private` as well since we don't want to touch them elsewhere.